graphs/frequent_pattern_graph_miner.py

```diff
@@ -0,0 +1,183 @@
+'''
+FP-GraphMiner - A Fast Frequent Pattern Mining Algorithm for Network Graphs
+
+A novel Frequent Pattern Graph Mining algorithm,
+FP-GraphMiner, that compactly represents a set of network graphs as a Frequent Pattern Graph (or FP-Graph).
+This graph can be used to efficiently mine frequent subgraphs including maximal frequent subgraphs and maximum common subgraphs.
+
+URL:https://www.researchgate.net/publication/235255851_FP-GraphMiner_-_A_Fast_Frequent_Pattern_Mining_Algorithm_for_Network_Graphs
+
+'''
+def get_DE(EA):
+    '''
+    Return Distinct edges from edge array of multiple graphs
+    '''
+    DE=set()
+    for i in range(len(EA)):
+        for j in range(len(EA[i])):
+            DE.add(EA[i][j][0])
+    de=list(DE)
+    return(de) 
+
+def get_bitcode(EA,DE):
+    '''
+    Return bitcode of DE
+    '''
+    bitcode=['0' for i in range(len(EA))] 
+    #bitcode="0" * len(EA)
+    for i in range(len(EA)):
+        for j in range(len(EA[i])):
+            if DE in EA[i][j][0]:
+                bitcode[i]='1'
+                break
+    return bitcode
+
+def get_FT(EA):
+    '''
+    Returns FT,cluster,nodes,support
+    '''
+    DE=get_DE(EA) 
+    FT=dict()
+    for i in range(len(DE)):
+        bit=get_bitcode(EA,DE[i])
+        bt=''.join(bit)
+        #print(bt)
+        s=bt.count('1')
+        FT[DE[i]]=[s,bt]
+    '''
+    Store [Distinct edge, WT(Bitcode), Bitcode] in Descending order
+    '''
+    Sorted_FT=[[k,v[0],v[1]] for k,v in sorted(FT.items(),key=lambda v:v[1][0],reverse=True)]

```

Comment:
Python variables are snake_case as discussed in CONTRIBUTING.md so `sorted_ft` but there is still the problem that ft is really unclear for the reader.